src/ResourceManagement/EventHub/DisasterRecoveryPairingAuthorizationRulesImpl.cs

```diff
@@ -0,0 +1,118 @@
+
+        IDisasterRecoveryConfigsOperations IHasInner<IDisasterRecoveryConfigsOperations>.Inner => this.Inner();
+
+        ///GENMHASH:62C00EEB1DF1BD51BEAC5339F5198000:2532064C6301665404E8957355D963BC
+        public IEnumerable<Microsoft.Azure.Management.Eventhub.Fluent.IDisasterRecoveryPairingAuthorizationRule> ListByDisasterRecoveryPairing(string resourceGroupName, string namespaceName, string pairingName)
+        {
+            return WrapList(Extensions.Synchronize(() => this.manager.Inner
+                            .DisasterRecoveryConfigs.ListAuthorizationRulesAsync(resourceGroupName,
+                                namespaceName,
+                                pairingName))
+                .AsContinuousCollection(nextLink => Extensions.Synchronize(() => this.manager.Inner
+                            .DisasterRecoveryConfigs.ListAuthorizationRulesNextAsync(nextLink))));
+        }
+
+        ///GENMHASH:29A726C59C9E8C7255A01C36C8AE6852:255DF06AA43E1C1CE6ED97FA24392A0F
+        public async Task<IPagedCollection<Microsoft.Azure.Management.Eventhub.Fluent.IDisasterRecoveryPairingAuthorizationRule>> ListByDisasterRecoveryPairingAsync(string resourceGroupName, string namespaceName, string pairingName, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            return await PagedCollection<IDisasterRecoveryPairingAuthorizationRule, AuthorizationRuleInner>
+                .LoadPage(
+                    async (cancellation) =>
+                        await this.manager.Inner
+                            .DisasterRecoveryConfigs.ListAuthorizationRulesAsync(resourceGroupName,
+                                namespaceName,
+                                pairingName,
+                                cancellation),
+                    async (nextLink, cancellation) => 
+                        await this.manager.Inner
+                            .DisasterRecoveryConfigs.ListAuthorizationRulesNextAsync(nextLink,
+                                cancellation),
+                    WrapModel, true, cancellationToken);
+        }
+
+        ///GENMHASH:B6961E0C7CB3A9659DE0E1489F44A936:168EFDB95EECDB98D4BDFCCA32101AC1
+        public EventHubManager Manager()
+        {
+            return this.manager;
+        }
+
+        ///GENMHASH:5002116800CBAC02BBC1B4BF62BC4942:37DAFCA0F979EB14168635F75681B9E4
+        public IDisasterRecoveryPairingAuthorizationRule GetById(string id)
+        {
+            return GetByIdAsync(id).Result;
+        }
+
+        ///GENMHASH:E776888E46F8A3FC56D24DF4A74E5B74:75BCE9962F8F51C054B48D3344BFE162
+        public Task<Microsoft.Azure.Management.Eventhub.Fluent.IDisasterRecoveryPairingAuthorizationRule> GetByIdAsync(string id, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            ResourceId resourceId = ResourceId.FromString(id);
+            return this.GetByNameAsync(resourceId.ResourceGroupName,
+                resourceId.Parent.Name,
```

Comment:
I think this method should be marked as async and this line should await.

Reply:
marking a method as `async` required only if we are invoking multiple async calls in the method body, if we add it anyway compiler is going to ignore it.

Reply:
But i can add them to ensure consistency.

Reply:
+1 I remember we talked about this in the past and we all agreed to use the async/await to ensure consistency and not because they are required. I believe one of the arguments was that it also makes it easier to distinguish between sync/async call event though looking at the return will have been enough.